substrate/primitives/keystore/src/testing.rs

```diff
@@ -309,7 +330,19 @@ impl Keystore for MemoryKeystore {
 		key_type: KeyTypeId,
 		seed: Option<&str>,
 	) -> Result<ecdsa_bls381::Public, Error> {
-		self.generate_new::<ecdsa_bls381::Pair>(key_type, seed)
+		let pubkey = self.generate_new::<ecdsa_bls381::Pair>(key_type, seed.clone())?;
+
+		let s: String = self.keys
+			.read()
+			.get(&key_type)
+			.map(|inner| inner.get(pubkey.as_slice()).map(|s| s.to_string()))
+			.flatten()
+			.expect("Can Retrieve Seed");
+
```

Comment:
```suggestion

// This is done to give the keystore access to individul keys, this is necessay to avoid unnecessary host functions for paired keys and re-use host functions implemented for each element of the pair.
```